sisyphus/job.py

```diff
@@ -220,11 +219,9 @@ def _sis_init(self, args, kwargs, parsed_args):
         self._sis_is_finished = False
         self._sis_setup_since_restart = False
 
-        self._sis_environment = None
-        if gs.CLEANUP_ENVIRONMENT:
-            self._sis_environment = tools.EnvironmentModifier()
-            self._sis_environment.keep(gs.DEFAULT_ENVIRONMENT_KEEP)
-            self._sis_environment.set(gs.DEFAULT_ENVIRONMENT_SET)
+        self._sis_environment = tools.EnvironmentModifier(cleanup_env=gs.CLEANUP_ENVIRONMENT)
+        self._sis_environment.keep(gs.DEFAULT_ENVIRONMENT_KEEP)
+        self._sis_environment.set(gs.DEFAULT_ENVIRONMENT_SET)
```

Comment:
Oh right. I thought about this, but then forgot again.

The user might still have set `DEFAULT_ENVIRONMENT_SET = {...sth...}` in its `settings.py`, and additionally also `CLEANUP_ENVIRONMENT = False`. (Actually, I myself have the settings like this, because I was playing around with the settings. Actually, I didn't really realized that `DEFAULT_ENVIRONMENT_SET` is not used when `CLEANUP_ENVIRONMENT = False`, I just wondered why it does not work, and thought I must have done sth wrong or there is a bug.) So just changing the default would still change the behavior in those cases, and I think we better should avoid to change the behavior.

So I only conditionally set those now. That should keep the old behavior.